### PR TITLE
track all paywall events on client and server

### PR DESCRIPTION
**Pull request by barshimi**
Close: #4744

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/.env [hunk 1]
```diff
@@ -9,4 +9,4 @@ NX_REACT_APP_FEATURE_FLAGS="{}"
 NX_REACT_APP_DATA_SOURCE="http://localhost:3000/graphql"
 NX_REACT_APP_STORAGE_GATEWAY_URL="http://localhost:3002"
 NX_REACT_APP_SERVER_URI=""
-NX_REACT_APP_BILLING_API_KEY="123456"
\ No newline at end of file
+NX_REACT_APP_BILLING_API_KEY="$2b$10$kvZ2zkq7onj/oyxH4vZLKuO1P3E9QKuRmo9z72pzggRtYWv4fkElC:415fb9ff-a378-42d0-a849-dae7daf49646"
```
Comment (abrl91) [inline packages/amplication-client/.env]: remove

File: packages/amplication-client/src/Entity/EntityList.tsx [hunk 2]
```diff
@@ -89,6 +91,12 @@ const EntityList: React.FC<Props> = ({ match, innerRoutes }) => {
     };
   }, [refetch, stopPolling, startPolling]);
 
+  const handleEntityClick = () => {
+    trackEvent({
+      eventName: AnalyticsEventNames.UpgradeOnEntityListClick,
```
Comment (abrl91) [inline packages/amplication-client/src/Entity/EntityList.tsx:96]: According to the Notion docs you need to add this metadata:
Reason: “Entities limit notification”
Location: App

File: packages/amplication-client/src/Workspaces/ResourceList.tsx [hunk 3]
```diff
@@ -45,6 +45,12 @@ function ResourceList() {
     setError(null);
   }, [setError]);
 
+  const handleResourceClick = () => {
+    trackEvent({
+      eventName: AnalyticsEventNames.UpgradeOnResourceListClick,
```
Comment (abrl91) [inline packages/amplication-client/src/Workspaces/ResourceList.tsx:50]: According to the Notion docs you need to add this metadata:
Reason: “Entities limit notification”
Location: App

File: packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.tsx [hunk 4]
```diff
@@ -86,6 +89,12 @@ const WorkspaceHeader: React.FC<{}> = () => {
     history.replace("/login");
   }, [history, apolloClient]);
 
+  const handleUpgradeClick = useCallback(() => {
+    trackEvent({
+      eventName: AnalyticsEventNames.UpgradeOnTopBarClick,
```
Comment (abrl91) [inline packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.tsx:94]: According to the Notion docs you need to add this metadata:
Reason: “Top panel CTA”
Location: App
Workspace: workspace id
User: user id

File: packages/amplication-server/src/core/billing/billing.service.ts [hunk 5]
```diff
@@ -110,6 +114,15 @@ export class BillingService {
         successUrl: new URL(cancelUrl, this.clientHost).href,
       },
     });
+    this.analytics.track({
+      userId: workspaceId,
+      event:
```
Comment (abrl91) [inline packages/amplication-server/src/core/billing/billing.service.ts:119]: I don't understand this event. You don't only have upgrade or downgrade. You also have a situation where the user didn't do anything

**Review by abrl91 — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Entity/EntityList.tsx (hunk 2 above)
Comment (barshimi) [reply packages/amplication-client/src/Entity/EntityList.tsx:96]: I can't add it. we need to move stigg provider to higher level in order to get the reason for the notification

File: packages/amplication-client/src/Workspaces/ResourceList.tsx (hunk 3 above)
Comment (barshimi) [reply packages/amplication-client/src/Workspaces/ResourceList.tsx:50]: I can't add it. we need to move stigg provider to higher level in order to get the reason for the notification

File: packages/amplication-server/src/core/billing/billing.service.ts (hunk 5 above)
Comment (barshimi) [reply packages/amplication-server/src/core/billing/billing.service.ts:119]: I don't understand.... this controller invoke when user click upgrade/downgrade his plan

Comment (barshimi) [reply packages/amplication-server/src/core/billing/billing.service.ts:119]: move stigg provider #4856

File: packages/amplication-client/src/Workspaces/ResourceList.tsx (hunk 3 above)
Comment (barshimi) [reply packages/amplication-client/src/Workspaces/ResourceList.tsx:50]: move stigg provider #4856

**Review by abrl91 — APPROVED** [review]

**Pull request merged**